pytorch_lightning/core/lightning.py

```diff
@@ -148,7 +148,7 @@ def optimizers(
             A single optimizer, or a list of optimizers in case multiple ones are present.
         """
         if use_pl_optimizer:
-            opts = list(self.trainer.lightning_optimizers.values())
+            opts = list(self.trainer.strategy._lightning_optimizers.values())
```

Comment:
I believe it was because of this test.

https://github.com/PyTorchLightning/pytorch-lightning/pull/11444#discussion_r783360396

But I don't know why we would want to support destroying the lightning optimizers inside the training step

Reply:
This is legacy behavior and should be removed. I believe this was meant to support dynamic optimizer, but we never added that feature.